src/main/java/seedu/taassist/logic/commands/CommandResult.java

```diff
@@ -74,16 +90,26 @@ public boolean equals(Object other) {
         }
 
         CommandResult otherCommandResult = (CommandResult) other;
-        return feedbackToUser.equals(otherCommandResult.feedbackToUser)
-                && showHelp == otherCommandResult.showHelp
-                && exit == otherCommandResult.exit
-                && focus == otherCommandResult.focus
-                && unfocus == otherCommandResult.unfocus;
+        boolean isEqualFeedback = feedbackToUser.equals(otherCommandResult.feedbackToUser);
+        boolean isEqualUiAction;
+        if (hasUiAction()) {
+            isEqualUiAction = uiAction.equals(otherCommandResult.uiAction);
+        } else {
+            isEqualUiAction = !otherCommandResult.hasUiAction();
+        }
```

Comment:
```suggestion
        boolean isEqualUiAction = uiAction == otherCommandResult.uiAction; 
```
`null` is not a problem for Enums.